util/unicode.h

```diff
@@ -61,6 +61,58 @@ constexpr std::string unicode_to_utf8(std::uint32_t code_point) {
     }
 }
 
+// https://infra.spec.whatwg.org/#surrogate
+constexpr bool is_unicode_surrogate(std::uint32_t code_point) {
```

Comment:
It's also already in in master, so this just moves it around and adds tests.

Since that PR is already quite large, I was going to clean up these functions existing in multiple places after it was merged, but since I ran into a bug here I had to write tests and this was the simplest way to deal with it.